map_test.go

```diff
@@ -2036,140 +2063,152 @@ func BenchmarkMap(b *testing.B) {
 }
 
 func BenchmarkIterate(b *testing.B) {
-	m, err := NewMap(&MapSpec{
-		Type:       Hash,
-		KeySize:    8,
-		ValueSize:  8,
-		MaxEntries: 1000,
-	})
-	if err != nil {
-		b.Fatal(err)
-	}
-	b.Cleanup(func() {
-		m.Close()
-	})
-
-	var (
-		n      = m.MaxEntries()
-		keys   = make([]uint64, n)
-		values = make([]uint64, n)
-	)
-
-	for i := 0; uint32(i) < n; i++ {
-		keys[i] = uint64(i)
-		values[i] = uint64(i)
-	}
-
-	_, err = m.BatchUpdate(keys, values, nil)
-	testutils.SkipIfNotSupported(b, err)
-	qt.Assert(b, qt.IsNil(err))
-
-	b.Run("MapIterator", func(b *testing.B) {
-		var k, v uint64
-
-		b.ReportAllocs()
-		b.ResetTimer()
-
-		for i := 0; i < b.N; i++ {
-			iter := m.Iterate()
-			for iter.Next(&k, &v) {
-				continue
-			}
-			if err := iter.Err(); err != nil {
-				b.Fatal(err)
+	for _, mt := range []MapType{Hash, PerCPUHash} {
+		m, err := NewMap(&MapSpec{
+			Type:       mt,
+			KeySize:    8,
+			ValueSize:  8,
+			MaxEntries: 1000,
+		})
+		if err != nil {
+			b.Fatal(err)
+		}
+		b.Cleanup(func() {
+			m.Close()
+		})
+		prefix := m.Type().String() + "/"
```

Comment:
Why not another level of `b.Run` nesting instead of manually adding `prefix`? Then filtering via `-bench` will also work correctly.